pkgs/development/compilers/polyml/default.nix

```diff
@@ -39,6 +46,11 @@ stdenv.mkDerivation rec {
     runHook postCheck
   '';
 
+  preInstall = ''
+    substituteInPlace polyc \
+      --replace-fail "LINK=\"$CXX\"" "LINK=\"$(type -P ${lib.getBin hostCC}/bin/c++)\""
+  '';
```

Comment:
We can do this in `postPatch` for `polyc.in` (replace `LINK=@CXX@`).
https://github.com/polyml/polyml/blob/a71e81c152470d53bdcaff767fb7305d6908a9a1/polyc.in#L5

Also we don’t need `type -P` in this case because we are not doing `PATH` lookup.

Reply:
> We can do this in postPatch for polyc.in (replace LINK=@CXX@).
https://github.com/polyml/polyml/blob/a71e81c152470d53bdcaff767fb7305d6908a9a1/polyc.in#L5

I prefer to do it in `preInstall`, because if `polyml` is using, or if in the future it starts using `polyc` during the build, then we want to use `$CXX`. It's only after `polyml` has finished building that we want `polyc` to use a different linker, i.e. one that runs in the runtime host rather than the build host.

> Also we don’t need type -P in this case because we are not doing PATH lookup.

Good point, I will fix this. Thanks!

Reply:
Also I think `lib.getExe'` is more appropriate in this case (it’s equivalent to `${getBin p}/bin/${name}` that we do here).